app/components/page_list_component/view.rb

```diff
@@ -57,5 +57,31 @@ def condition_page_position(condition)
       check_page = @pages.find { |page| page.id == condition.check_page_id }
       page_position(check_page)
     end
+
+    def conditions_for_page_with_index(page_id)
+      routing_conditions_with_index.fetch(page_id, [])
+    end
+
+    def routing_conditions_with_index
+      @routing_conditions_with_index ||= process_routing_conditions
+    end
+
```

Comment:
process_routing_conditions is a fun method!

We are labelling the conditions with order of their check_page_id but displaying them in order of routing_page_id. We use routing_conditions_with_index as an index to conditions, memoising it so that we only do it once for the page list.

It's a hash. The keys are page_id's, the values are arrays of [condition, index].

When we display conditions and we need to create names like "Route n of question m's route" the index is the n.

I'm very open to suggestions for better ways to do this.

Reply:
Thanks for the detailed explanation of what this is doing and the challenges.

I think this is a good approach in general. If we wanted to be able to test it more it might be worth splitting it out into its own class somewhere, but if we're not expecting to reuse this anywhere that might be overkill.

Reply:
Yes, I think this code might end up in it's own service so that the numbering is in one place. I think that passing in the conditions for the page might simplify this code.

I just want to check that this is used elsewhere, probably in making new routes.